### PR TITLE
MINIFICPP-1544 Add Custom Text property to GenerateFlowFile processor

### PR DESCRIPTION
**Pull request by lordgamez**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/standard-processors/processors/GenerateFlowFile.cpp [hunk 1]
```diff
@@ -61,6 +62,13 @@ core::Property GenerateFlowFile::UniqueFlowFiles(
     core::PropertyBuilder::createProperty("Unique FlowFiles")->withDescription("If true, each FlowFile that is generated will be unique. If false, a random value will be generated and all FlowFiles")
         ->isRequired(false)->withDefaultValue<bool>(true)->build());
 
+core::Property GenerateFlowFile::CustomText(
+    core::PropertyBuilder::createProperty("Custom Text")
+      ->withDescription("If Data Format is text and if Unique FlowFiles is false, then this custom text will be used as content of the generated FlowFiles and the File Size will be ignored. "
+                        "Finally, if Expression Language is used, evaluation will be performed only once per batch of generated FlowFiles")
+      ->supportsExpressionLanguage(true)
```
Comment (adamdebreceni) [inline extensions/standard-processors/processors/GenerateFlowFile.cpp:69]: we could emit a warning when Custom Text is set but it won't be used

Comment (arpadboda) [reply extensions/standard-processors/processors/GenerateFlowFile.cpp:69]: +1

File: extensions/standard-processors/processors/GenerateFlowFile.cpp [hunk 2]
```diff
@@ -110,6 +119,15 @@ void GenerateFlowFile::onSchedule(const std::shared_ptr<core::ProcessContext> &c
     logger_->log_trace("Unique Flow files is configured to be %i", uniqueFlowFile_);
   }
 
+  if (textData_ && !uniqueFlowFile_) {
+    std::string custom_text;
+    context->getProperty(CustomText, custom_text, nullptr);
+    if (!custom_text.empty()) {
+      std::copy(custom_text.begin(), custom_text.end(), std::back_inserter(data_));
```
Comment (arpadboda) [inline extensions/standard-processors/processors/GenerateFlowFile.cpp:126]:
```
data_.assign(custom_text.begin(), custom_text.end());
```

For me this feels like a bit shorter and more efficient, but I may missed some constraints here when using this.

**Review by arpadboda — COMMENTED** [review]
Mostly looks good, two minor comments.

File: extensions/standard-processors/processors/GenerateFlowFile.cpp (hunk 2 above)
Comment (lordgamez) [reply extensions/standard-processors/processors/GenerateFlowFile.cpp]: Updated in 8b3702dd930f7b38cc744e3e1917b6c7380816b8

File: extensions/standard-processors/processors/GenerateFlowFile.cpp (hunk 1 above)
Comment (lordgamez) [reply extensions/standard-processors/processors/GenerateFlowFile.cpp:68]: Added warning and test in 8b3702dd930f7b38cc744e3e1917b6c7380816b8

**Review by arpadboda — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**